content/en/blog/_posts/Windows-Memory-Pressure-Eviction.md

```diff
@@ -0,0 +1,85 @@
+
+2. **Paging (swap) is automatically enabled on Windows**, and it plays an essential role in the operating system's memory management. Disabling paging is not recommended because it can lead to system instability.
+
+## Eviction Thresholds
+
+By default, Windows nodes have a default "hard" eviction threadhold of `--eviction-hard=memory.available<500Mi`. This means that if the system's availalbe memory is less than 500 Mi the Kubelet will try and start evicting pods.
+
+For more information on configuring memory-pressure based eviction and understanding the eviction signals and threadhols, refer to the official Kubernetes documentation on [memory-pressure-eviction](https://kubernetes.io/docs/concepts/scheduling-eviction/node-pressure-eviction/#eviction-signals-and-thresholds).
+
+
+## Monitoring Commit Memory for Windows nodes
+
+Understanding the distinction between **Working Set** and **Commit** memory and being able to monitor both is critical for maintaining the stability of individual Pod workloads and the entire Windows node.
+
+Currently, `kubectl top` continues to report Working Set memory. As a result, a Pod may be evicted for exceeding its memory requests or encounter memory allocation failures due to exceeding its Commit limit, even when the WorkingSet appears to be well within the expected bounds. This discrepancy can lead to confusion.
+
+Starting with v1.31, Windows nodes now report the global `CommitTotal` and `CommitLimit` for each node through the `/stats/summary` API endpoint. This information is available under the `windows-global-commit-memory` syscontainer.
+
+```json=
+kubectl get --raw http://{cluster-endpoint}/api/v1/nodes/{node-name}/proxy/stats/summary
+
+{
+ "node": {
+  "nodeName": "{node-name},
+  "systemContainers": [
+   ...
+   {
+    "name": "windows-global-commit-memory",
+    "startTime": "2024-10-02T19:14:33Z",
+    "memory": {
+     "time": "2024-10-02T19:14:33Z",
+     "availableBytes": 4563402752,
+     "usageBytes": 1212754432
+    }
+   }
+  ...
+  ],
+}
+
+```
+
+
+Additionally, exposing the Commit usage for individial Pods and Containers through the `/stats/summary` API is being worked on as part of the [cAdvisor-les, CRI-full Container and Pod Stats](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2371-cri-pod-container-stats/README.md) enhancement.
+
+[windows-exporter](https://github.com/prometheus-community/windows_exporter) can also be configured to monitor and report on Commit memory usage.
+
+## Additional Resources
+
+To gain a deeper understanding of how memory is managed on Windows, you may also find this article on [Physical and Virtual Memory in Windows 10](https://answers.microsoft.com/en-us/windows/forum/all/physical-and-virtual-memory-in-windows-10/e36fb5bc-9ac8-49af-951c-e7d39b979938) helpful.
```

Comment:
Given that Windows 11 is available, does this Windows 10 article from 2015 apply to the newer version also? Or is it scoped only to 10, and maybe 11 has some other info that we need to point to?

Reply:
It is still relevant and is the most informative article I could find on the topic

Reply:
That is one detailed blog post 😮